### DISCUSSION TITLE
Retrieving logs from a model (not causer)

### DISCUSSION THREAD

**Discussion by Yentel** [spatie/laravel-activitylog#800] [Q&A] [answered]
Is there an opposite to the CausesActivity trait?

The documentation shows on how you can log an activity and then request it using the CausesActivity trait.
For example: `activity()->performedOn($newsArticle)->causedBy($user)`.
I can then request all activities by using $user->actions.

However, is this also possible when starting from the model (i.e. newsarticle)?
I am trying to get a list of all actions taken by any user on a certain model to show a history of what happened to it since it has been created.

Thanks for creating this package, and the many other Spatie packages, they are amazing, piece by piece.

**Comment by Gummibeer** [comment, marked as answer]
Hey, the trait included is the `Spatie\Activitylog\Traits\LogsActivity` one which also provides automatic model change logging.
https://spatie.be/docs/laravel-activitylog/v3/advanced-usage/logging-model-events

If you only want the relationship just copy the relationship method.
```php
public function activities(): MorphMany
{
    return $this->morphMany(ActivitylogServiceProvider::determineActivityModel(), 'subject');
}
```
